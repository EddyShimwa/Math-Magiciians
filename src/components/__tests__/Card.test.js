import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Card from '../interface/Card';

describe('Compare the Card component current output to a previous snapshot', () => {
  test('The component output should be consistent with the previous snapshot.', () => {
    const card = renderer
      .create(
        <Card>
          <h1>Hello from Card Component</h1>
        </Card>,
      )
      .toJSON();

    expect(card).toMatchSnapshot();
  });

  test('The component should be rendered and visible on the page at all times.', () => {
    render(
      <Card>
        <h1>This is component</h1>
      </Card>,
    );

    expect(screen.getByText('This is component')).toBeInTheDocument();
  });
});
