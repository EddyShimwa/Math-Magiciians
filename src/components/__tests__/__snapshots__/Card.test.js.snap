// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare the Card component current output to a previous snapshot The component output should be consistent with the previous snapshot. 1`] = `
<div
  className="card"
>
  <h1>
    Hello from Card Component
  </h1>
</div>
`;
